module/main.go

```diff
@@ -39,3 +54,29 @@ func mainWithArgs(ctx context.Context, args []string, logger golog.Logger) error
 	<-ctx.Done()
 	return nil
 }
+
+func printVersion(args []string, logger golog.Logger) Arguments {
+	var argsParsed Arguments
+	// Don't propagate error if there are additional flags
+	// that are not recognized. Otherwise the module fails
+	// under normal operation.
+	//nolint:errcheck
+	_ = utils.ParseFlags(args, &argsParsed)
```

Comment:
[opt] This really should be directly in mainWithArgs (it's really strange that printVersion returns parsed CLI arguments.) You can/should also return an error there instead of leaving it unchecked.

But... I know this program likely does some weird stuff with cli args later on, and it's ONLY the version argument for now, so considering this optional for now.